### PR TITLE
test(filter): add unit test of filter

### PR DESCRIPTION
**Pull request by Zippo-Wang**
**What this PR does / why we need it**:
add unit test of filter

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestFilterGetQ
--- PASS: TestFilterGetQ (0.00s)
=== RUN   TestFilterData
--- PASS: TestFilterData (0.00s)
=== RUN   TestFilterFrom
--- PASS: TestFilterFrom (0.00s)
=== RUN   TestFilterGetFrom
--- PASS: TestFilterGetFrom (0.00s)
=== RUN   TestFilterGetQueries
--- PASS: TestFilterGetQueries (0.00s)
=== RUN   TestFilterWhere
--- PASS: TestFilterWhere (0.00s)
=== RUN   TestFilterSlice
--- PASS: TestFilterSlice (0.00s)
=== RUN   TestFilterJson
--- PASS: TestFilterJson (0.00s)
=== RUN   TestToStrSlice
--- PASS: TestToStrSlice (0.00s)
=== RUN   TestContains
--- PASS: TestContains (0.00s)
=== RUN   TestHas
--- PASS: TestHas (0.00s)
=== RUN   TestHasContain
--- PASS: TestHasContain (0.00s)
PASS

coverage: 93.0% of statements in ../../../../terraform-provider-huaweicloud/...
```

### CODE REVIEW

File: huaweicloud/helper/filters/filter_test.go [hunk 1]
```diff
@@ -162,3 +162,289 @@ func TestFilter2(t *testing.T) {
...
+	queriesData1 = append(queriesData1, queriesData2...)
+	assert.NotNil(t, filter.GetQueries())
+	assert.Equal(t, queriesData1, filter.GetQueries())
+}
+
+func TestFilterWhere(t *testing.T) {
+	filter := New()
+	key1, operator1, value1 := "foo1", ">", "bar1"
+	key2, operator2, value2 := "foo2", "<", "bar2"
+	key3, operator3, _ := "foo3", "=", "bar3"
```
Comment (deer-hang) [inline huaweicloud/helper/filters/filter_test.go]:
```suggestion
	key3, operator3 := "foo3", "="
```

Comment (Zippo-Wang) [reply huaweicloud/helper/filters/filter_test.go]: fix

**Pull request merged**